plugins/module_utils/foreman_helper.py

```diff
@@ -26,6 +26,34 @@
     APYPIE_IMP_ERR = traceback.format_exc()
 
 
+def _entity_spec_helper(spec):
+    """Extend an entity spec by adding entries for all alt_ and flat_names.
+    Extract ansible compatible argument_spec on the way.
+    """
+    entity_spec = {'id': {}}
+    argument_spec = {}
+    for key, value in spec.items():
+        if 'alt_names' in value:
+            value = value.copy()
+            alt_names = value.pop('alt_names')
```

Comment:
What's the benefit of copying and then popping a value?

Reply:
The reason for copying before changing is, that those dictionaries in Python are always passed around by value. So if i do not copy it, i might change a dictionary that was already assigned for another purpose in entity_spec, which i should not.
Maybe this first occurance of copy is not _desperately needed_, but i think it is a good custom not to change the `spec` variable that was passed originally.